pkg/resource/configurator.go

```diff
@@ -75,3 +75,28 @@ func kindish(obj runtime.Object) string {
 	}
 	return strings.ToLower(reflect.TypeOf(obj).Elem().Name())
 }
+
+type ObjectMetaConfiguratorWithNewNamingFormat struct {
```

Comment:
I can definitely use some help with the name of that struct..

Reply:
I say we just change the existing configurator then. In both cases, the default would be the new scheme and people who want to use the old one can supply their own configurator.